### ISSUE TITLE
training_step bug

### ISSUE THREAD

**Issue by linsole** [FzJun26th/FastRecon#3]
Hi, thank you for your code! I want to point out that during training step, self.embedding_list and self.embedding_list_mu_cor are overwritten at the end of function, so technichally only the last batch of calculated results are saved in class member variable. I believe that is a bug, bacause the implementation in https://github.com/hcw-00/PatchCore_anomaly_detection/blob/main/train.py#L284C22-L284C22 actually extend the member variable not overwrite it.

**Comment by Zhangwt95** [comment]
The open source code doesn't seem complete enough. Have you compared the precisions between extending and overwriting the member variable.

**Comment by linsole** [comment]
> The open source code doesn't seem complete enough. Have you compared the precisions between extending and overwriting the member variable.

No, I wasn't able to run the code successfully. I found this bug when debugging, and I didn't make it eventually.

**Comment by CharlesChen24** [comment]
I have the same problem
